File: frontend/packages/ai/src/prompt/templates/before-after/instruction.ts
export function BeginningOfText(): string {
  return `
    You are tasked with suggesting a compelling start to the user's text. Your goal is to provide a natural and contextually appropriate beginning that seamlessly flows into the existing content.
    `;
}

export function EndOfText(): string {
  return `
    You must predict the next words the user will write. Only predict as many words as approrpriate.
    `;
}

export function MiddleOfText(): string {
  return `You are an expert writer tasked with seamlessly connecting two parts of a sentence or paragraph. Your goal is to provide a natural and contextually appropriate bridge that maintains the flow and coherence of the text.`;
}

export function InstructionBeforeAfter(before: string, after: string): string {
  if (before.length === 0) {
    return BeginningOfText();
  } else if (after.length === 0) {
    return EndOfText();
  } else {
    return MiddleOfText();
  }
}

File: backend/packages/prompt/src/templates/predict/instruction.py
def beginning_of_text() -> str:
    return """
    You are an AI assistant that helps authors with great introductions.
    """

def end_of_text() -> str:
    return """
    You are a prediction assistant to an author. Help the author figure out the next words to write.
    """

def middle_of_text() -> str:
    return """
    You are tasked with suggesting revisions to the passage to improve clarity, flow, and style.
    """

def SuggestInstruction(before: str, after: str) -> str:
    if len(before) == 0:
        return f"{beginning_of_text()}"
    elif len(after) == 0:
        return f"{end_of_text()}"
    else:
        return f"{middle_of_text()}"
